### PR TITLE
Maximum invites and invites are stored

### PR DESCRIPTION
**Pull request by pitaj**
Added the ability for admins to restrict the maximum amount of invites
a user can make. Invites are stored and displayed in the registration
queue admin page.

### CODE REVIEW

**Comment by julianlam** [comment]
Wasn't aware we had the ability to have users invite users at all, does this PR introduce that functionality?

**Comment by pitaj** [comment]
@julianlam Nope. It was already there. Should I add another toggle to disable non-admin user invites?

**Comment by julianlam** [comment]
Yes, that is my expectation, that we'd probably want to have either admin-only invites, or admin+user invites.

Could it not be specified in the "Registration type" dropdown instead?

**Comment by pitaj** [comment]
Yeah, it could be. 

Adding `Admin Invite Only` to the dropdown could work. Let me see.

File: src/controllers/users.js [hunk 1]
```diff
@@ -150,14 +150,26 @@ usersController.getUsersForSearch = function(req, res, next) {
...
+	plugins.fireHook('filter:users.build', { req: req, res: res, templateData: data }, function(err, data) {
 		if (err) {
 			return next(err);
 		}
 
-		data.templateData.inviteOnly = meta.config.registrationType === 'invite-only';
-		data.templateData['reputation:disabled'] = parseInt(meta.config['reputation:disabled'], 10) === 1;
-		res.render('users', data.templateData);
+		if (!req.uid) {
+			return next(new Error('[[error:no-privileges]]'));
```
Comment (barisusakli) [inline src/controllers/users.js]: this will prevent guest users from seeing the user list?

Comment (pitaj) [inline src/controllers/users.js]: Yeah. Oops. I'll remove that.

**Comment by barisusakli** [comment]
So this gives each user X invites and once they use that up they won't be able to invite anyone else right? At least thats what I think it does from glancing over the code.

**Comment by pitaj** [comment]
@barisusakli Exactly. And Admins get infinite invites.

File: src/controllers/users.js [hunk 2]
```diff
@@ -150,14 +150,26 @@ usersController.getUsersForSearch = function(req, res, next) {
...
+		}
+
+		user.getInvitesNumber(req.uid, function(err, num) {
```
Comment (barisusakli) [inline src/controllers/users.js]: make sure `getInvitesNumber` returns 0 if uid is 0

Comment (pitaj) [inline src/controllers/users.js]: It returns null

**Comment by pitaj** [comment]
Alright, added another entry to the dropdown. The templates in persona, etc should be updated so in the case of `inviteOnly`, it only shows the invite button when users are logged in. In the case of `adminInviteOnly`, it only shows the button if the user is an admin.

**Comment by pitaj** [comment]
Alright, we now got NodeBB/nodebb-theme-persona#211

**Comment by julianlam** [comment]
To be merged post 0.9.2

**Pull request merged**